source/B2BApi/Functions/EnqueueMessages/BRS_045/EnqueueHandler_Brs_045_MissingMeasurementsLog.cs

```diff
@@ -25,11 +28,35 @@ public class EnqueueHandler_Brs_045_MissingMeasurementsLog(
     private readonly IOutgoingMessagesClient _outgoingMessagesClient = outgoingMessagesClient;
     private readonly IUnitOfWork _unitOfWork = unitOfWork;
 
-    public Task HandleAsync(
+    public async Task HandleAsync(
         EnqueueMissingMeasurementsLogHttpV1 missingMeasurementsLog,
         CancellationToken cancellationToken)
     {
-        // TODO: Implement enqueueing outgoing RSM-018 message.
-        return Task.CompletedTask;
+        foreach (var message in missingMeasurementsLog.Data)
+        {
+            var outgoingMessageDto = CreateOutgoingMessageDto(message);
+
+            await _outgoingMessagesClient.EnqueueAsync(
+                outgoingMessageDto,
+                cancellationToken).ConfigureAwait(false);
+        }
+
+        await _unitOfWork.CommitTransactionAsync(cancellationToken).ConfigureAwait(false);
+    }
+
+    private MissingMeasurementMessageDto CreateOutgoingMessageDto(EnqueueMissingMeasurementsLogHttpV1.DateWithMeteringPointId data)
```

Comment:
```suggestion
    private MissingMeasurementMessageDto CreateOutgoingMessageDto(EnqueueMissingMeasurementsLogV1.DateWithMeteringPointId data)
```

I'm not sure we using protocol on DTO models?

Reply:
If we are going to remove the protocol. Then EnqueueCalculatedMeasurementsHttpV1 should be updated aswell

Reply:
Lets keep the name for now.